src/LibChorus/VcsDrivers/Mercurial/HgRepository.cs

```diff
@@ -600,13 +601,35 @@ private ExecutionResult Execute(bool failureIsOk, bool noChangeIsOk, int seconds
 			return result;
 		}
 
+		/// <summary>
+		/// The procedure for recoverring from an interrupted update, is to update again.
+		/// </summary>
+		private ExecutionResult RecoverFromInterruptedUpdate(bool failureIsOk, int secondsBeforeTimeout, string cmd, string[] rest)
+		{
+			Update();
+			return Execute(failureIsOk, secondsBeforeTimeout, cmd, rest);
+		}
+
 		private ExecutionResult RecoverFromFailedMerge(bool failureIsOk, int secondsBeforeTimeout, string cmd, string[] rest)
 		{
-			_progress.WriteMessageWithColor(@"Blue", "Attempting to recover from failed merge.");
-			var result = Execute(false, SecondsBeforeTimeoutOnMergeOperation, "resolve", "--all");
-			if (!String.IsNullOrEmpty(result.StandardError))
+			var heads = GetHeads();
+			if(heads.Count() > 2)
+			{
+				throw new ApplicationException(String.Format("Unable to recover from failed merge: [Too many heads in the repository]"));
+			}
+			MergeSituation.PushRevisionsToEnvironmentVariables(heads[0].UserId, heads[0].Number.Hash,
```

Comment:
Well, the short answer is that the test for this situation failed in Mercurial 3 so I was forced to evaluate it and I discovered that it probably didn't ever work at all. Pushing to the environment is required for our merge tool to be used during the recovery attempt. Our merge tool was not actually used in the recovery attempt before, so whenever this happened historically it almost certainly corrupted data.